hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -72,13 +72,19 @@ public HiveCatalog(Configuration conf) {
   }
 
   public HiveCatalog(String name, String uri, int clientPoolSize, Configuration conf) {
+    this(name, uri, null, clientPoolSize, conf);
+  }
+
+  public HiveCatalog(String name, String uri, String warehouse, int clientPoolSize, Configuration conf) {
     this.name = name;
     this.conf = new Configuration(conf);
     // before building the client pool, overwrite the configuration's URIs if the argument is non-null
     if (uri != null) {
       this.conf.set(HiveConf.ConfVars.METASTOREURIS.varname, uri);
     }
-
+    if (warehouse != null) {
+      this.conf.set(HiveConf.ConfVars.METASTOREWAREHOUSE.varname, warehouse);
```

Comment:
Why not keep this location as a field in the `HiveCatalog` instead of storing it in `conf`? Then the default logic would only need to be run once.

Reply:
As we discussed in this [comment](https://github.com/apache/iceberg/pull/1586/files#r509462458),   we could set `warehouse` string or `hive-conf-dir` to get the hive warehouse.   If use a local field,  then in getWarehouseLocation,  we would check the local field first and then read the hiveConf ?  I want to unify the parse path so I  put this key-value into hiveConf (It's a deep-cloned conf, so should not affect the origin conf).

Reply:
Yes, It can be handled in the same way as URIs.

BTW, can you unify `hive.metastore.warehouse.dir` and `HiveConf.ConfVars.METASTOREWAREHOUSE.varname`?

Reply:
I think handling this the same way that we handle the URI makes sense. We set the URI because we need to pass it back into Hive to connect, which I don't think applies to the warehouse path, but handling both the same way is reasonable.